### DISCUSSION TITLE
Extend mbsubmit plugin to directly open MB add release with pre-populated fields

### DISCUSSION THREAD

**Discussion by jonaswinkler** [beetbox/beets#5299] [Ideas]
Hello! Dev question. I'd like to enhance the mbsubmit plugin. The new "open picard" option is very nice, but I'd like to go further and add an option to directly open the MB /add/release page in the user's browser, with known metadata already filled (so you wouldn't need to have Picard installed for that).

Similar to how "Print tracks" prints the tracks of an unmatched album for copy/paste and similar to how "Open Picard" launches Picard with the files to be added to MB, this new option (Maybe "Submit to [M]usicbrainz") would open a new browser window/tab, and the add release forms will be populated with known metadata.

This is entirely possible, MB Picard does it as well, by serving a small web page with a built-in web server that auto POSTs form data to musicbrainz on load, and opening that web page in a browser. See https://github.com/metabrainz/picard/blob/master/picard/browser/addrelease.py. Conveniently also written in python.

This is official, documentation for seeding the release editor is found here: https://musicbrainz.org/doc/Development/Seeding/Release_Editor

Since we can redirect to localhost after adding a release, it is also entirely possible to automatically process the new release in beets with the mbid returned by musicbrainz.

Questions:

1. Would this be a good feature?
2. For implementation I'd re-use and adapt relevant parts from the picard code base. Are there licensing issues with that?

I considered to just import picard and call the relevant code parts, but

A) the code might change since they don't expect other components to call it directly,
B) it does not really work without the entire picard application running, and
C) it pulls in lots of UI-related dependencies.

**Comment by jonaswinkler** [comment]
Prototype is working smooth. PR soon if you want it.

**Reply by arsaboo** [reply]
I think it will be a great addition.
